contracts/drafts/ERC1820Implementer.sol

```diff
@@ -0,0 +1,22 @@
+pragma solidity ^0.5.2;
+
+import "./IERC1820Implementer.sol";
+
+/**
+ * @dev ERC1820Implementer allows your contract to implement an interface for another account in the sense of ERC1820.
+ * That account or one of its ERC1820 managers can register the implementer in the ERC1820 registry, but the registry
+ * will first check with the implementer if it agrees to it, and only allow it if it does.
+ */
+contract ERC1820Implementer is IERC1820Implementer {
+    bytes32 constant private ERC1820_ACCEPT_MAGIC = keccak256(abi.encodePacked("ERC1820_ACCEPT_MAGIC"));
+
+    mapping(bytes32 => mapping(address => bool)) private _supportedInterfaces;
+
+    function canImplementInterfaceForAddress(bytes32 interfaceHash, address account) external view returns (bytes32) {
+        return _supportedInterfaces[interfaceHash][account] ? ERC1820_ACCEPT_MAGIC : bytes32(0x00);
+    }
+
+    function _registerInterfaceForAddress(bytes32 interfaceHash, address account) internal {
```

Comment:
I understand that you used `Address` in the function name to be consistent with the EIP function, but this is kind of inconsistent with the name `account` that we're using. What are your thoughts on this?

Reply:
I'd rather keep `Address` in the function name to make it clearer that it refers to the same addresses as the EIP. I don't think the name of the function argument is that important to be honest, its meaning should be clear from the function name itself.